opensearch/src/main/java/org/opensearch/sql/opensearch/response/OpenSearchResponse.java

```diff
@@ -91,7 +94,21 @@ public Iterator<ExprValue> iterator() {
       }).iterator();
     } else {
       return Arrays.stream(hits.getHits())
-          .map(hit -> (ExprValue) exprValueFactory.construct(hit.getSourceAsString())).iterator();
+          .map(hit -> {
+            ExprValue docData = exprValueFactory.construct(hit.getSourceAsString());
+            if (hit.getHighlightFields().isEmpty()) {
+              return docData;
+            } else {
+              ImmutableMap.Builder<String, ExprValue> builder = new ImmutableMap.Builder<>();
+              builder.putAll(docData.tupleValue());
+              var hlBuilder = ImmutableMap.<String, ExprValue>builder();
+              for (var es : hit.getHighlightFields().entrySet()) {
```

Comment:
especially for one-liners, you can consider using the lambda forEach syntax:
```
              hit.getHighlightFields().entrySet().forEach(
                      es -> hlBuilder.put(es.getKey(), ExprValueUtils.stringValue(es.getValue().toString()))
              );
```
applies elsewhere